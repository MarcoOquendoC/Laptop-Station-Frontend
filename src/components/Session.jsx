import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { login } from '../redux/Registration/auth';

const Session = () => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const dispatch = useDispatch();

  const handleSubmit = (e) => {
    e.preventDefault();
    const user = {
      email,
      password,
    };
    dispatch(login({ user }));
  };

  return (
    <>
      <div>
        <form onSubmit={handleSubmit}>
          <label htmlFor="email">
            <input onChange={(e) => setEmail(e.target.value)} id="email" type="email" placeholder="email" />
          </label>
          <label htmlFor="password">
            <input onChange={(e) => setPassword(e.target.value)} id="password" type="password" placeholder="password" />
          </label>
          <button type="submit">Log in</button>
        </form>
        <p>
          Or
          <Link to="/signup">Sign up</Link>
          if you don&apos;t have an account.
        </p>
      </div>
    </>
  );
};

export default Session;
