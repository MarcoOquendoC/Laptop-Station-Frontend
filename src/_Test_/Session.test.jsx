import '@testing-library/jest-dom';
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { screen, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import Session from '../components/Session';
import store from "../redux/configureStore";

describe('Testing Home', () => {
  let login;
  beforeAll(async () => {
    login = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Session />
          </Router>
        </Provider>,
      )
      .toJSON();
  });

  it('should renders correctly', () => {
    expect(login).toBeTruthy();
  });

  it('should match the snapshot', () => {
    expect(login).toMatchSnapshot();
  });

  it('renders the email input field', () => {
    const emailInput = waitFor(() => screen.getByPlaceholderText('email'));
    waitFor(() => expect(emailInput).toBeInTheDocument());
    waitFor(() => expect(emailInput.type).toBe('email'));
    waitFor(() => expect(emailInput.required).toBe(true));
  });

  it('renders the password input field', () => {
    const passwordInput = waitFor(() => screen.getByPlaceholderText('password'));
    waitFor(() => expect(passwordInput).toBeInTheDocument());
    waitFor(() => expect(passwordInput.type).toBe('password'));
    waitFor(() => expect(passwordInput.required).toBe(true));
  });
});