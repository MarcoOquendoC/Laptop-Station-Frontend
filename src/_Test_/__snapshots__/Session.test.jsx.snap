// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Home should match the snapshot 1`] = `
Array [
  <div
    className="principal"
  >
    <div
      className="header"
    >
      <div
        className="menu mob"
      >
        <img
          alt="menu"
          src="menu.svg"
        />
      </div>
      <div
        className="menu_logo"
      >
        <img
          alt="logo"
          className="logo"
          src="logo_laptop_station.svg"
        />
      </div>
      <div
        className="menu mob"
      />
    </div>
    <nav
      className="menunav"
    >
      <ul
        className="menunav_ul"
      >
        <li
          className="menu_text"
        >
          <a
            href="/login"
            onClick={[Function]}
          >
            Log in
          </a>
        </li>
        <li
          className="menu_text select"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="menu_text"
        >
          <a
            href="/laptops"
            onClick={[Function]}
          >
            Latops
          </a>
        </li>
        <li
          className="menu_text"
        >
          <a
            href="/reserves"
            onClick={[Function]}
          >
            Reserves
          </a>
        </li>
        <li
          className="menu_text admin"
        >
          <a
            className="a__admin"
            href="/addItem"
            onClick={[Function]}
          >
            Add item
          </a>
        </li>
        <li
          className="menu_text"
        />
      </ul>
    </nav>
  </div>,
  <div
    className="viewfinder"
  >
    <div
      className="register"
    >
      <h1>
        Log In
      </h1>
      <form
        className="form"
        onSubmit={[Function]}
      >
        <input
          className="form__input"
          onChange={[Function]}
          placeholder="email"
          type="email"
        />
        <input
          className="form__input"
          onChange={[Function]}
          placeholder="password"
          type="password"
        />
        <div
          className="form__input"
        />
        <button
          className="form__btn"
          type="submit"
        >
          Log in
        </button>
      </form>
      <p>
        Or
        <a
          className="orsignup"
          href="/signup"
          onClick={[Function]}
        >
           Sign up 
        </a>
        if you don't have an account.
      </p>
    </div>
  </div>,
]
`;
